src/StrStr.java

```diff
@@ -0,0 +1,31 @@
+public class StrStr {
+    public static int strStr(String haystack, String needle) {
+        char[] haystackChars = haystack.toCharArray();
```

Comment:
You do not need to convert string to char arrays. Solve this problem in O(1) memory.

Reply:
Got it. I'll get chars by charAt() method